samples/algorithms/sudoku-grover/Program.cs

```diff
@@ -0,0 +1,474 @@
+                if (good)
+                    Console.WriteLine("quantum result verified correct");
+                Pause();
+            }
+            Console.WriteLine("finished");
+
+        }
+
+        // An Empty square with its row and column
+        public class EmptySquare
+        {
+            public int i, j;
+        }
+
+        /// # Summary
+        /// Copy an Int 2 dimensional array
+        ///
+        /// # Input
+        /// ## org
+        /// The array to copy
+        /// ## size
+        /// The size of the array
+        ///
+        /// # Output
+        /// A copy of the array
+        public static int[,] CopyIntArray(int[,] org, int size)
+        {
+            int[,] result = new int[size, size];
+            for (int i = 0; i < size; i++)
+                for (int j = 0; j < size; j++)
+                    result[i, j] = org[i, j];
+            return result;
+        }
+ 
+        /// # Summary
+        /// QuantumSolve will call Q# code to solve the Sudoku puzzle and display the solution
+        /// # Input
+        /// ## puzzle
+        /// The Sudoku puzzle to solve
+        /// ## size
+        /// The size of the puzzle is either 4 (for 4x4) or 9 (for 9x9)
+        /// ## subsize
+        /// subsize is 2 for 4x4 and 3 for 9x9
+        /// ## sim
+        /// The quantum simulator
+        public static void QuantumSolve(int[,] puzzle, int size, int subSize, QuantumSimulator sim)
+        {
+            List<ValueTuple<long, long>> emptySquareEdges;
+            HashSet<ValueTuple<long, long>> startingNumberConstraints;
+            List<EmptySquare> emptySquares;
+            FindEdgesAndInitialNumberConstraints(puzzle, size, subSize, out emptySquareEdges, out startingNumberConstraints, out emptySquares);
+            var emptySquareEdgesQArray = new QArray<ValueTuple<long, long>>(emptySquareEdges);
+            var startingNumberConstraintsQArray = new QArray<ValueTuple<long, long>>(startingNumberConstraints);
+            Console.WriteLine("Quantum solving puzzle ");
+            ShowGrid(puzzle, size);
+            var task = SolvePuzzle.Run(sim, emptySquares.Count, size, emptySquareEdgesQArray, startingNumberConstraintsQArray);
+            if (!task.Result.Item1)
+                Console.WriteLine("no solution found ");
+            else
+            {
+                var solution = task.Result.Item2.ToArray();
+                for (int i = 0; i < solution.Length; i++)
+                {
+                    puzzle[emptySquares[i].i, emptySquares[i].j] = (int)solution[i] + 1;
+                }
+                Console.WriteLine("solved puzzle ");
+                ShowGrid(puzzle, size);
+            }
+        }
+
+        /// # Summary
+        /// Display the puzzle
+        public static void ShowGrid(int[,] puzzle, int size)
+        {
+            for (int i = 0; i < size; i++)
+            {
+                for (int j = 0; j < size; j++)
```

Comment:
It's generally helpful to make sure that `for` loops are formatted uniformly; having a mix of single line (no `{}`) and block style (with `{}`) makes it hard to tell at a glance what's included in each loop, especially since the various loops aren't indented uniformly (276 and 288 use different indentation widths).